docs/data/xrootd/install-shoveler.md

```diff
@@ -0,0 +1,128 @@
+```
+
+Installing the Shoveler
+-----------------------
+
+The shoveler can be installed via RPM, Container, or staticly compiled binary.
+
+Requirements for running the Shoveler
+
+1. An open port (configurable) that can receive UDP packets from the XRootD servers.  It does not need to be an open port to the internet, only open to the XRootD servers.
+2. Outgoing TCP connectivity.
+3. A directory to store the on-disk queue.
+
+Configuring the Shoveler
+------------------------
+
+Configuration can be specified with environment variables or a configuration file.  The configuration file is in `yaml`.  An example configuration file is distributed with the shoveler.  In the RPM, the configuration file is located in `/etc/xrootd-monitoring-shoveler/config.xml`.  Below, we will break the configuration file into fragments, but they together make a whole configuration file.
+
+Environment variables can be derived from the yaml.  Every environment variable starts with `SHOVELER_`, then continues with the structure of the configuration file.  For example, the `amqp url` can be configured with the environment variable `SHOVELER_AMQP_URL`.  The `verify` option can be configured with `SHOVELER_VERIFY`.
+
+### Configuration Fragments
+
+```yaml
+# AMQP configuration
+amqp:
+  url: amqps://username:password@example.com/vhost
+  exchange: shoveled-xrd
+  topic:
+  token_location: /etc/xrootd-monitoring-shoveler/token
+```
+
+AMQP configuration.  For the OSG, the url should be `amqps://clever-turkey.rmq.cloudamqp.com/xrd-mon`.  The `exchange` should is correct for the OSG.  `token_location` is the path to the authentication token.
+
+```yaml
+listen:
+  port: 9993
+  ip: 0.0.0.0
+```
+
+Where to listen for UDP messages from XRootD servers.
+
+```yaml
+verify: true
+```
+
+Whether to verify the header of the packet matches XRootD's monitoring packet format.
+
+```yaml
+# Export prometheus metrics
+metrics:
+  enable: true
+  port: 8000
+```
+
+Listening location of [Prometheus](https://prometheus.io/) metrics to view the performance and status of the shoveler in Prometheus format.
+
+```yaml
+config_directory: /tmp/shoveler-queue
+```
+Directory to store overflow of queue onto disk. The queue keeps 100 messages in memory.  If the shoveler is disconnected from the message bus, it will store messages over the 100 in memory onto disk into this directory.  Once the connection has been re-established the queue will be emptied.  The queue on disk is persistent between restarts.
+
+```yaml
+# map:
+#   all: 172.0.0.4
+```
+Mapping configuration (optional). If `map.all` is set, all messages will be mapped to the configured IP address.
+For example, with the above configuration, if a packet comes in with the private IP address of 192.168.0.4, the packet origin will be changed to 172.0.0.4.  The port is always preserved.
+
+```yaml
+# map:
+#   192.168.0.5: 172.0.0.5
+#   192.168.0.6: 129.93.10.7
+```
+If you want multiple mappings, you can specify multiple map entries.
```

Comment:
I think each of these fragments should live under a `#### <foo>` subsection: it'll be easier to find in the ToC on the RHS. Also, our docs generally have the description first then the example so we should follow that